maker-bundle/src/Resources/config/hooks.yaml

```diff
@@ -0,0 +1,566 @@
+
+    compileFormFields:
+        return_type: array
+        arguments:
+            fields: array
+            formId: string
+            form: Contao\Form
+
+    createDefinition:
+        return_type: array
+        arguments:
+            key: string
+            value: string
+            definition: string
+            '&dataSet': array
+
+    createNewUser:
+        return_type: void
+        arguments:
+            userId: int
+            userData: array
+            module: Contao\Module
+
+    customizeSearch:
+        return_type: void
+        arguments:
+            '&pageIds': array
+            keywords: string
+            queryType: string
+            fuzzy: bool
+            module: Contao\Module
+
+    executePostActions:
+        return_type: void
+        arguments:
+            action: string
+            dc: Contao\DataContainer
+
+    executePreActions:
+        return_type: void
+        arguments:
+            action: string
+
+    executeResize:
+        return_type: ?string
+        arguments:
+            image: Contao\Image
+
+    exportTheme:
+        return_type: void
+        arguments:
+            xml: DomDocument
+            zipArchive: Contao\ZipWriter
+            themeId: int
+
+    extractThemeFiles:
+        return_type: void
+        arguments:
+            xml: DomDocument
+            zipArchive: Contao\ZipReader
+            themeId: int
+            mapper: array
+
+    generateBreadcrumb:
+        return_type: array
+        arguments:
+            items: array
+            module: Contao\Module
+
+    generateFrontendUrl:
+        return_type: string
+        arguments:
+            page: array
+            params: string
+            url: string
+
+    generatePage:
+        return_type: void
+        arguments:
+            pageModel: Contao\PageModel
+            layout: Contao\LayoutModel
+            pageRegular: Contao\PageRegular
+
+    generateXmlFiles:
+        return_type: void
+        arguments: []
+
+    getAllEvents:
+        return_type: array
+        arguments:
+            events: array
+            calendars: array
+            timeStart: int
+            timeEnd: int
+            module: Contao\Module
+
+    getArticle:
+        return_type: void
+        arguments:
+            article: Contao\ArticleModel
+
+    getArticles:
+        return_type: ?string
+        arguments:
+            pageId: int
+            column: string
+
+    getAttributesFromDca:
+        return_type: array
+        arguments:
+            attributes: array
+            dc: [Contao\DataContainer, 'null']
+
+    getCombinedFile:
+        return_type: string
+        arguments:
+            content: string
+            key: string
+            mode: string
+            file: array
+
+    getContentElement:
+        return_type: string
+        arguments:
+            contentModel: Contao\ContentModel
+            buffer: string
+            contentElement: Contao\ContentElement
+
+    getCountries:
```

Comment:
Some of these hooks are already deprecated. Might be helpful if we can show that in the maker too IMO.
(But maybe in a seperate pull request)